Kernel/VM/PhysicalRegion.h

```diff
@@ -0,0 +1,35 @@
+#pragma once
+
+#include <AK/Bitmap.h>
+#include <AK/Retainable.h>
+#include <AK/Retained.h>
+#include <Kernel/PhysicalAddress.h>
+#include <Kernel/VM/PhysicalPage.h>
+
+class PhysicalRegion : public Retainable<PhysicalRegion> {
+    AK_MAKE_ETERNAL
+
+public:
+    static Retained<PhysicalRegion> create(PhysicalAddress lower, PhysicalAddress upper);
+    ~PhysicalRegion() {}
+
+    PhysicalAddress lower() { return m_lower; }
+    PhysicalAddress upper() { return m_upper; }
+    unsigned size() const { return m_pages; }
+    unsigned used() const { return m_used; }
+    unsigned free() const { return m_pages - m_used; }
+    bool owns_page(PhysicalPage& page) const { return page.paddr() >= m_lower && page.paddr() <= m_upper; }
+
+    PhysicalAddress take_free_page();
```

Comment:
I think it would be nicer to have
```cpp
RetainPtr<PhysicalPage> take_free_page();
```
That way the code clearly communicates that the caller is receiving something that he now part-owns.